addons/website/tests/test_qweb.py

```diff
@@ -107,3 +107,86 @@ def test_process_att_url_crap(self):
             match.reset_calls()
             self._test_att('/x?y#z', {'href': '/x?y#z'})
             match.assert_called_with('/x', method='POST', query_args='y')
+
+class TestQwebDataSnippet(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.env['ir.ui.view'].create({
+            'name': 'some_html',
+            'type': 'qweb',
+            'key': 'website.some_html',
+            'arch': '''
+                <t t-name="some_html"><article><span>Hello</span><t t-out="0"/></article></t>
+            '''
+        })
+
+        self.env['ir.ui.view'].create({
+            'name': 's_a',
+            'type': 'qweb',
+            'key': 'website.s_a',
+            'arch': '''
+                <t t-name="s_a"><section class="hello"><t t-call="website.some_html"/><t t-out="0"/></section></t>
+            '''
+        })
+        self.env['ir.ui.view'].create({
+            'name': 's_b',
+            'type': 'qweb',
+            'key': 'website.s_b',
+            'arch': '''
+                <t t-name="s_b"><section class="foo"><t t-snippet-call="website.s_a"/></section></t>
+            '''
+        })
+        self.env['ir.ui.view'].create({
+            'name': 's_c',
+            'type': 'qweb',
+            'key': 'website.s_c',
+            'arch': '''
+                <t t-name="s_c"><t t-call="website.some_html"><p>World!</p></t></t>
+            '''
+        })
+        self.env['ir.ui.view'].create({
+            'name': 's_d',
+            'type': 'qweb',
+            'key': 'website.s_d',
+            'arch_db': '''
+                <t t-name="s_d"><t t-snippet-call="website.s_a"><p>World!</p></t></t>
+            '''
+        })
+    
+    def _render_snippet(self, snippet):
+        render_template = self.env['ir.ui.view'].create({
+            'name': f't-snippet-call_{snippet}',
+            'type': 'qweb',
+            'arch': f'''
+                <t t-snippet-call="{snippet}"/>
+            '''
+        })
+        return self.env['ir.qweb']._render(render_template.id)
+
+    def test_case_a(self):
```

Comment:
Seems good to name the snippet a, b, c, d but for the test name, make it more descriptive please :pray: Something like `test_t_call_inside_snippet`, `test_t_call_as_snippet_root`, ...